contracts/staking/StakingInfo.sol

```diff
@@ -189,11 +197,13 @@ contract StakingInfo {
         emit Staked(
             signer,
             validatorId,
+            validatorNonce[validatorId].add(1),
             activationEpoch,
             amount,
             total,
             signerPubkey
         );
+        validatorNonce[validatorId] = validatorNonce[validatorId].add(1);
```

Comment:
We should do it before `emit`, so we don't have to add twice. For all events.

Reply:
yes, I'll change it.

Reply:
done